File: Codes/0344-reverse-string.cpp
/*
 * @Author: LetMeFly
 * @Date: 2022-10-01 09:31:35
 * @LastEditors: LetMeFly
 * @LastEditTime: 2022-10-01 09:31:36
 */
#ifdef _WIN32
#include "_[1,2]toVector.h"
#endif

class Solution {
public:
    void reverseString(vector<char>& s) {
        int n = s.size();
        for (int i = 0; i < n / 2; i++) {
            swap(s[i], s[n - i - 1]);
        }
    }
};

File: Codes/0344-reverse-string.py
'''
Author: LetMeFly
Date: 2023-08-07 07:57:36
LastEditors: LetMeFly
LastEditTime: 2023-08-07 07:58:37
'''
from typing import List

class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        for i in range(len(s) // 2):
            s[i], s[len(s) - i - 1] = s[len(s) - i - 1], s[i]